src/docker.rs

```diff
@@ -101,17 +79,50 @@ pub fn run(target: &Target,
     // We create/regenerate the lockfile on the host system because the Docker
     // container doesn't have write access to the root of the Cargo project
     let cargo_toml = root.join("Cargo.toml");
+
+    let mut runner = None;
+    let mut container_engine = DOCKER.to_string();
```

Comment:
I think we should use the `which` crate to automatically detect whether `docker` or `podman` is installed.

Reply:
I'm fine with this, but I have a concern for users who may have both installed. What should `cross` default to in these cases?

Reply:
I'd say default to Docker since that's the current behaviour.